plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -1330,6 +1360,89 @@ private static <K, V> Cache<K, AtomicReference<V>> buildBulkCache(
         return cacheBuilder.build();
     }
 
+    private class TableNamesCache
+    {
+        private static final String DUMMY_CACHE_KEY = "$KEY";
+
+        @Nullable
+        private final LoadingCache<String, List<String>> perSchemaCache;
+        /**
+         * Normally, we would use a memoized supplier. However, there is an additional refresh
+         * capability so we use a single entry loading cache with a dummy key
+         */
+        @Nullable
+        private final LoadingCache<String, Map<String, List<String>>> batchCache;
+
+        private final boolean isBatchCache;
+
+        public TableNamesCache(
+                CacheFactory cacheFactory,
+                com.google.common.base.Function<String, List<String>> perSchemaLoader,
+                Supplier<Map<String, List<String>>> batchLoader)
+        {
+            isBatchCache = delegate.supportBatchGetTables();
+
+            if (isBatchCache) {
+                batchCache = cacheFactory.buildCache((com.google.common.base.Function<String, Map<String, List<String>>>) (ignore -> batchLoader.get()));
```

Comment:
IntelliJ recomendation - casting can be skipped.

Reply:
Thanks, It appears that I had my InteliJ warnings screwed up.